raft/raft_flow_control_test.go

```diff
@@ -50,8 +50,8 @@ func TestMsgAppFlowControlFull(t *testing.T) {
 	for i := 0; i < 10; i++ {
 		r.Step(pb.Message{From: 1, To: 1, Type: pb.MsgProp, Entries: []pb.Entry{{Data: []byte("somedata")}}})
 		ms := r.readMessages()
-		if len(ms) != 0 {
-			t.Fatalf("#%d: len(ms) = %d, want 0", i, len(ms))
+		if len(ms) != 1 || len(ms[0].Entries) != 0 {
```

Comment:
This means that now every proposal triggers an empty `MsgApp` when Inflights is saturated. Previously it would only be after a heartbeat response.

This is probably because `maybeSendAppend`/`sendAppend` is invoked from a few other places. We may need to restrict this, to minimize the number of empty appends.

Reply:
Done.